src/test/scala/org/bitcoins/core/util/Base58Test.scala

```diff
@@ -53,5 +90,53 @@ class Base58Test extends FlatSpec with MustMatchers {
     Base58.encode(Base58.decode(multi)) must be (bitcoinj)
   }
 
+  it must "read base58_keys_valid.json and validate each case" in {
+    import org.bitcoins.core.util.testprotocol.Base58ValidTestCaseProtocol._
+    val source = Source.fromURL(this.getClass.getResource("/base58_keys_valid.json"))
+    val lines = try source.getLines.filterNot(_.isEmpty).map(_.trim) mkString "\n" finally source.close()
+    val json = lines.parseJson
+    val testCases : Seq[Base58ValidTestCase] = json.convertTo[Seq[Base58ValidTestCase]]
+    for {
+      testCase <- testCases
+    } yield {
+      testCase must be (Base58ValidTestCaseImpl(testCase.addressOrWIFPrivKey, testCase.hashOrPrivKey, testCase.configParams))
+      if (testCase.addressOrWIFPrivKey.isLeft) {
+        Base58.isValid(testCase.addressOrWIFPrivKey.left.get.value) must be (true)
+      }
+      else println("Base58.isValid(testCase.addressOrWIFPrivKey.right.get): " + Base58.isValid(testCase.addressOrWIFPrivKey.right.get))
+    }
+
+    //first, second and 48th test cases:
+    testCases.head must be (Base58ValidTestCaseImpl(Left(Address("1AGNa15ZQXAZUgFiqJ2i7Z2DPU2J6hW62i")),
+      Left(Sha256Hash160Digest("65a16059864a2fdbc7c99a4723a8395bc6f188eb")),
+      ConfigParamsImpl(Left("pubkey"), false, false)))
+
+    testCases(1) must be (Base58ValidTestCaseImpl(Left(Address("3CMNFxN1oHBc4R1EpboAL5yzHGgE611Xou")),
+      Left(Sha256Hash160Digest("74f209f6ea907e2ea48f74fae05782ae8a665257")), ConfigParamsImpl(Left("script"), false, false)))
+
+    testCases(47) must be (Base58ValidTestCaseImpl(Right("cMxXusSihaX58wpJ3tNuuUcZEQGt6DKJ1wEpxys88FFaQCYjku9h"),
+      Right(ECPrivateKey("0b3b34f0958d8a268193a9814da92c3e8b58b4a4378a542863e34ac289cd830c")),
+      ConfigParamsImpl(Right(true), true, true)))
+  }
+
+  it must "read base58_keys_invalid.json and return each as an invalid base58 string" in {
+    import org.bitcoins.core.util.testprotocol.Base58InvalidTestCase
+    import org.bitcoins.core.util.testprotocol.Base58InvalidTestCaseProtocol._
+
+    val source = Source.fromURL(this.getClass.getResource("/base58_keys_invalid.json"))
+    val lines = try source.getLines.filterNot(_.isEmpty).map(_.trim) mkString "\n" finally source.close()
+    val json = lines.parseJson
+    val testCases : Seq[Base58InvalidTestCase] = json.convertTo[Seq[Base58InvalidTestCase]]
+    for {
+      testCase <- testCases
+    } yield {
+      testCase must be (Base58InvalidTestCaseImpl(testCase.base58EncodedString))
+      Base58.isValid(testCase.base58EncodedString) match {
```

Comment:
This is not right, you can replace this with `Base58.isvalid(...) must be (false)` instead of your match statement